### PR TITLE
2 QoL thingies

### LINKED ISSUE #6092: Add Player Ping To Player List
image

-Add each players ping to the server, to the player list tab.

-Also update and make UI cleaner

### LINKED ISSUE #6971: Putting your ID card in your PDA could use text feedback.
### Description
Title. Putting your ID card in your PDA doesn't create a message in chat stating as such. This might confuse new players when they use their ID card on their PDA and it seemingly dissappears.

### PR DESCRIPTION
**Pull request by MaxIsJoe**
closes #6971
closes #6092 ~~(Testing is appreciated)~~

CL: [Improvement] placing ID inside PDA has text feedback.
CL: [Improvement] player list now shows player ping.

### CODE REVIEW

File: UnityProject/Assets/Scripts/Managers/PlayerList.cs [hunk 1]
```diff
@@ -575,6 +575,7 @@ public struct ClientConnectedPlayer
 {
 	public string UserName;
 	public string Tag;
+	public int pingToServer;
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Managers/PlayerList.cs]:
```suggestion
	public int PingToServer;
```
Think we want PascelCase

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Managers/PlayerList.cs]: Done 👍

**Comment by ThatDan123** [comment]
Still need to sort out a way to refresh the ping as it only happens when a new player connects. We forgot about this :p

Either server resends the data every so often (easiest)

Or the the client asks the server every so often when they have the playerlist open to refresh

**Comment by MaxIsJoe** [comment]
Oh yeah forgot about that, will work on it asap.

> Still need to sort out a way to refresh the ping as it only happens when a new player connects. We forgot about this :p
> 
> Either server resends the data every so often (easiest)
> 
> Or the the client asks the server every so often when they have the playerlist open to refresh

File: UnityProject/Assets/Scripts/UI/Systems/IngameMenu/LobbyUIPlayerListController.cs [hunk 2]
```diff
@@ -41,8 +41,8 @@ public void GenerateList()
 			var c = playerEntry.GetComponent<LobbyUIListTemplate>();
 			c.playerName.text = $"{player.Tag} {player.UserName}";
 			c.playerNumber.text = i.ToString();
+			c.playerPing.text = player.PingToServer >= -1 ? $"{player.PingToServer * 1000}ms" : "NaN";
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/UI/Systems/IngameMenu/LobbyUIPlayerListController.cs]:
```suggestion
			c.playerPing.text = player.PingToServer > 0 ? $"{player.PingToServer * 1000}ms" : "NaN";
```

**Comment by MaxIsJoe** [comment]
Please put a "needs testing" label on this.

I need someone (or two people) to host a server **online** and let others join to them and test the ping label to see whats wrong with it.

**Comment by Bod9001** [comment]
I don't mind hosting send me a ping

**Comment by MaxIsJoe** [comment]
Tested with Bod, works like a charm 👍 

We can merge now.

**Comment by ThatDan123** [comment]
did you test on headless?

File: UnityProject/Assets/Scripts/UI/Systems/IngameMenu/LobbyUIPlayerListController.cs [hunk 3]
```diff
@@ -41,9 +41,25 @@ public void GenerateList()
...
+			UpdatePingList();
+			GenerateList();
+			StartCoroutine(RefreshPing(list));
+		}
+	}
+
+	private void UpdatePingList()
+	{
+		Messages.Server.UpdateConnectedPlayersMessage.Send();
+	}
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/UI/Systems/IngameMenu/LobbyUIPlayerListController.cs]: This refresh will only work on local host when they have the player list open

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/UI/Systems/IngameMenu/LobbyUIPlayerListController.cs]: Fixed 👍

**Comment by MaxIsJoe** [comment]
we can remove the "needs testing" and "dnm" label now

File: UnityProject/Assets/Scripts/UI/Systems/IngameMenu/LobbyUIPlayerListController.cs [hunk 4]
```diff
@@ -1,11 +1,12 @@
 ﻿using System.Collections;
 using System.Collections.Generic;
 using System;
+using Mirror;
 using UnityEngine;
 using TMPro;
 using ServerInfo;
 
-public class LobbyUIPlayerListController : MonoBehaviour
+public class LobbyUIPlayerListController : NetworkBehaviour
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/UI/Systems/IngameMenu/LobbyUIPlayerListController.cs]: We dont want to make UI components Network Behaviours, you'll need to make a custom network message.

**Review by Bod9001 — DISMISSED** [review]

**Review by ThatDan123 — CHANGES_REQUESTED** [review]
https://github.com/unitystation/unitystation/pull/7764#discussion_r745478825

File: UnityProject/Assets/Scripts/UI/Systems/IngameMenu/LobbyUIPlayerListController.cs (hunk 4 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/UI/Systems/IngameMenu/LobbyUIPlayerListController.cs]: like this? https://github.com/unitystation/unitystation/pull/7764/commits/b2bd322e501cf223034a9fde1eaa0f2c81d84197

**Comment by ThatDan123** [comment]
Yeah, only issue is if multiple people have it open then they all will force the list to update at different times. 

Probably wont be an issue as the list update message isnt a performance drain.

**Review by ThatDan123 — APPROVED** [review]

**Review by ThatDan123 — CHANGES_REQUESTED** [review]
One last thing, you need to go back to the prefab with the LobbyUIPlayerListController component on and remove the network identity that was added when it was a network behaviour. Might as well do the same for the signal manager prefab in this PR too.

**Comment by MaxIsJoe** [comment]
> One last thing, you need to go back to the prefab with the LobbyUIPlayerListController component on and remove the network identity that was added when it was a network behaviour. Might as well do the same for the signal manager prefab in this PR too.

I'm one step ahead of you and I've already deleted it
image

**Comment by ThatDan123** [comment]
> > One last thing, you need to go back to the prefab with the LobbyUIPlayerListController component on and remove the network identity that was added when it was a network behaviour. Might as well do the same for the signal manager prefab in this PR too.

I am blind, gonna blame it on the fact ive just woken up

**Review by ThatDan123 — APPROVED** [review]

**Review by Bod9001 — APPROVED** [review]

**Pull request merged**